hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -1308,12 +1305,11 @@ public void performTableServices(Option<String> inFlightInstantTimestamp) {
         return;
       }
       // Check and run clean operations.
-      String latestDeltacommitTime = lastInstant.get()
-          .getTimestamp();
-      LOG.info("Latest deltacommit time found is " + latestDeltacommitTime + ", running clean operations.");
-      cleanIfNecessary(writeClient, latestDeltacommitTime);
+      cleanIfNecessary(writeClient);
```

Comment:
within cleanIfNecessary, we reload the active timelne. which we already did at L1302. lets revisit these calls and ensure we are reloading the timeline only when required

Reply:
yeah, but reusing is easy to cause bugs, maybe a better solution is we improve the performance of timeline loading.